### PR TITLE
topo: add lexically stabilised topological sort

### PR DESCRIPTION
**Pull request by kortschak**
@btracey Please take a look.

### CODE REVIEW

File: topo/tarjan.go [hunk 1]
```diff
@@ -31,24 +31,46 @@ func (e Unorderable) Error() string {
...
 // listing cyclic components in g with each cyclic component's members sorted by ID. When
 // an Unorderable error is returned, each cyclic component's topological position within
 // the sorted nodes is marked with a nil graph.Node.
 func Sort(g graph.Directed) (sorted []graph.Node, err error) {
 	sccs := TarjanSCC(g)
-	sorted = make([]graph.Node, 0, len(sccs))
+	return finalizeSort(sccs, lexical)
+}
+
+// SortStabilized performs a topological sort of the directed graph g returning the 'from'
```
Comment (btracey) [inline topo/tarjan.go]: I would add something along the lines of "order, when called, should re-order the nodes in-place".

File: topo/tarjan.go [hunk 2]
```diff
@@ -31,24 +31,46 @@ func (e Unorderable) Error() string {
...
+// position within the sorted nodes is marked with a nil graph.Node.
+func SortStabilized(g graph.Directed, order func([]graph.Node)) (sorted []graph.Node, err error) {
+	if order == nil {
+		order = lexical
+	}
+	sccs := tarjanSCCstabilized(g, order)
+	return finalizeSort(sccs, order)
+}
+
+func finalizeSort(sccs [][]graph.Node, order func([]graph.Node)) ([]graph.Node, error) {
```
Comment (btracey) [inline topo/tarjan.go]: I don't understand why this is called `finalizeSort`. Wouldn't something like `sscToSort` be a better name?

Comment (kortschak) [inline topo/tarjan.go]: `sortedFrom`

File: topo/tarjan.go (hunk 1 above)
Comment (kortschak) [inline topo/tarjan.go]: Done

**Comment by btracey** [comment]
LGTM

**Pull request merged**